### ISSUE TITLE
Confidence-score model for resolve_incident: calibration harness, per-tenant weights, threshold (epic #99)

### ISSUE THREAD

**Issue by 100rd** [100rd/Omniscience#155]
Parent: #99

## Blocked by

- The `resolve_incident` MCP tool sub-issue in epic #99 (Wave 2). v0.1 of `resolve_incident` ships a deliberately simple equal-weight scoring; this sub-issue replaces the fixed weights with a calibrated model.

## Blocks

- Closing epic #99 with confidence-score acceptance signed-off, IF the team decides v0.1 fixed-weight scoring is insufficient for the demo. Otherwise this is a follow-up.

## Context

`resolve_incident` (Wave 2 sub-issue) returns recommendations ranked by a confidence score in `[0, 1]`. v0.1 of that tool ships a fixed-weight formula:

```
confidence = 0.25*recency + 0.25*graph_proximity + 0.25*evidence_count + 0.25*cross_ref_strength
```

Equal weights are not a tuned model. They are a placeholder that lets the tool ship without blocking on calibration data. **Vision §11 explicitly flags this**: "Identity resolution accuracy. Confidence scores below a trust threshold produce wrong answers just as readily as bad data. Threshold tuning is an empirical, design-partner-driven task."

This sub-issue replaces the fixed weights with a calibrated model and produces a calibration-data harness that tracks score quality over time. It is **the second design-partner-feedback gate** for the incident-resolution path (the first being the demo scenario fixture itself).

The architect memo on #99 punted the formula to this sub-issue (or to v0.1 of `resolve_incident`, depending on how thin the v0.1 ships). State up-front in the design: this is a follow-up tuning task, not a v0.1 blocker.

## Scope

1. **Calibration dataset format**: a JSONL file at `tests/fixtures/incident_calibration.jsonl` with one record per labelled incident:
   ```json
   {
     "alert_id": "...",
     "tenant_id": "...",
     "recommendations": [
       {"target": "...", "evidence_ids": [...], "ground_truth_correct": true, "human_confidence": 0.9}
     ],
     "labelled_at": "...",
     "labelled_by": "..."
   }
   ```
   Loader logic that consumes this file to produce a per-component score-vs-truth comparison.
2. **Calibration harness** in `tests/test_resolve_incident_calibration.py`:
   - Replays each labelled incident through `resolve_incident` (without the calibration weights, in a "raw" mode that returns the per-component scores).
   - Computes per-component correlation with the ground-truth label.
   - Outputs a calibration report (CSV / Markdown) summarising AUC, precision@1, precision@3 for each component AND for the combined formula.
3. **Tuned weights**: fit a logistic regression (or simpler — even a coordinate-descent on AUC for 4 weights is fine and more interpretable) over the labelled dataset. Output: a single `IncidentScoringWeights` Pydantic model with fields `recency`, `graph_proximity`, `evidence_count`, `cross_ref_strength` that sum to 1.0. v0.2 weights replace the fixed equal weights in `resolve_incident`.
4. **Per-tenant calibration support** (architect memo open question): the `IncidentScoringWeights` model carries a `tenant_id: uuid.UUID | None` field. When `None`, weights are global defaults. When set, weights are per-tenant. The `resolve_incident` tool looks up the most-specific weights for the calling workspace and falls back to global. Storage: a new Postgres table `incident_scoring_weights` with one row per (tenant_id, weights_version). Schema migration via Alembic.
5. **Threshold recommendation**: from the calibration data, recommend a `confidence_threshold` for the demo recommendations: scores below the threshold are returned as "low confidence" with a warning flag in the response. Threshold can be per-tenant, same model as weights.
6. **`omniscience-admin calibrate` CLI command** that takes a calibration JSONL, fits weights, optionally writes them to the DB. Read-only by default; `--write-db --tenant-id <uuid>` actually persists. Same auth model as other admin CLI commands.
7. **Telemetry**: Prometheus histogram `omniscience_resolve_incident_confidence_score` (already named in the v0.1 sub-issue). This sub-issue adds counters for `omniscience_resolve_incident_below_threshold_total{tenant_id}`.

## Non-goals

- **A trained ML model**. Logistic regression over 4 hand-engineered features. Anything richer (gradient-boosted trees, neural ranker) is a separate sub-issue once we have ≥1000 labelled incidents per design partner — we won't reach that for months.
- **Online learning**. Weights are fitted offline via the CLI command. Online updating from production traffic is out of scope.
- **Cross-tenant data sharing**. Each tenant's labelled data stays in their workspace; the global weights are fitted from the union ONLY when explicit cross-tenant aggregation is approved (out of scope for v0.1).
- **Per-alert-class calibration** (alert from PagerDuty vs. Datadog vs. internal). Single weights vector per tenant in v0.1; per-class is a follow-up if/when calibration data warrants the parameter explosion.
- **Replacing the v0.1 formula structure**. The four components stay; only their weights and the threshold are calibrated.

## ACL invariant

- Calibration data is workspace-scoped. The `incident_scoring_weights` table has `tenant_id` as a non-null column for tenant-specific weights, NULL for global. The `resolve_incident` tool reads the row matching the caller's `workspace_id`, falling back to NULL.
- The calibration JSONL files are workspace-scoped: file naming and loader convention enforce that one file = one tenant's data. Cross-tenant aggregation is an explicit, out-of-band ops step.
- The CLI command `omniscience-admin calibrate --write-db` requires explicit `--tenant-id` and refuses to write global weights without `--allow-global` (a footgun guard).
- Cross-workspace isolation regression test: tenant A's labelled data NEVER influences tenant B's per-tenant weights.

## Dependencies

- **Reads**: `resolve_incident` (Wave 2 sub-issue) — must be stable before this sub-issue's calibration harness runs.
- **Reads**: existing admin CLI machinery (`omniscience-admin` command pattern).
- **Adds dependency on**: `scikit-learn` ONLY if the implementer chooses logistic-regression over a hand-rolled coordinate-descent. The hand-rolled path keeps zero new deps. **Decision deferred to the implementer.**
- **Schema migration**: one new Alembic migration for the `incident_scoring_weights` table.

## Acceptance

1. **Calibration harness runs** on a seeded calibration JSONL fixture (≥10 labelled incidents) and produces a CSV / Markdown report.
2. **Weights fit reproducibly** — same JSONL, same random seed, same fitted weights.
3. **Per-tenant override works** — when a tenant has a row in `incident_scoring_weights`, `resolve_incident` uses those weights; otherwise falls back to global defaults; otherwise falls back to the v0.1 equal weights (defensive).
4. **Threshold recommendation** is included in the calibration report and persisted alongside the weights.
5. **`omniscience-admin calibrate` CLI** loads, fits, and (with `--write-db`) persists weights. Round-trip test in CI.
6. **Cross-workspace isolation regression test** passes (described in ACL section).
7. **No breaking change to `resolve_incident`'s response shape** — the per-recommendation `confidence` field stays a float in `[0, 1]`. New fields are additive (`below_threshold: bool` flag).
8. **Documented in `docs/decisions/`** — a follow-up ADR (number assigned at PR time) committing to the formula structure, the calibration approach, and the per-tenant override mechanism. The ADR is required because this fixes a public contract (the score scale) for design partners.

## Labels

`feat`, `mcp`, `retrieval`, `observability`
